scala3doc/test/dotty/dokka/diagram/HierarchyTest.scala

```diff
@@ -0,0 +1,98 @@
+                    )
+                }
+                if (x.getName == "E2") {
+                    assertEquals(List("A1", "A2[Int]", "A3[Int, String]", "Any", "B1", "B2", "B3", "C1[Int, Boolean, Any]", "D2[Int, Boolean]", "D3", "Object"), x.getParentsAsStrings)
+                    assertEquals(List("C1[Int, Boolean, Any]", "D2[Int, Boolean]", "D3"), x.getDirectParentsAsStrings)
+                    assertEquals(List.empty, x.getKnownChildrenAsStrings)
+                    val graph = MemberExtension.getFrom(x).map(_.graph)   
+                    assertEquals(true, graph.isDefined)
+                    assertEquals(
+                        Seq(
+                            "Object" -> "Any",
+                            // "A1" -> "Object", // These are not applicable beacuase of bug and its workaround 
+                            // "A2[Int]" -> "Object", // More info at ClassLikeSupport.scala:37
+                            // "A3[Int, String]" -> "Object",
+                            // "B1" -> "Object",
+                            // "B1" -> "A1",
+                            // "B2" -> "Object",
+                            // "B2" -> "A1",
+                            // "B2" -> "A2[Int]",
+                            // "B3" -> "Object",
+                            // "B3" -> "A2[Int]",
+                            // "B3" -> "A3[Int, String]",
+                            // "C1[Int, Boolean, Any]" -> "Object",
+                            // "C1[Int, Boolean, Any]" -> "B1",
+                            // "C1[Int, Boolean, Any]" -> "B2",
+                            // "C1[Int, Boolean, Any]" -> "B3",
+                            "E2" -> "D2[Int, Boolean]",
+                            "E2" -> "D3",
+                            "D2[Int, Boolean]" -> "Object",
+                            "D3" -> "Object",
+                            "E2" -> "C1[Int, Boolean, Any]"
+                        ).sorted,
+                        graph.get.edges.map((a, b) => (a.signature.getName, b.signature.getName)).sorted
+                    )
+                }
+                if (x.getName == "A2") {
+                    assertEquals(List("Any", "Object"), x.getParentsAsStrings)
+                    assertEquals(List.empty, x.getDirectParentsAsStrings)
+                    assertEquals(List("B2", "B3", "C1[A, B, C]", "E1", "E2"), x.getKnownChildrenAsStrings)
+                    val graph = MemberExtension.getFrom(x).map(_.graph)   
+                    assertEquals(true, graph.isDefined)
+                    assertEquals(
+                        Seq(
+                            "Object" -> "Any",
+                            "A2[T]" -> "Object",
+                            "B2" -> "A2[T]", // These are not actually true, becuase we lose information about hierarchy in subtypes and their possible mapping to supertypes other that that type itself, e. g. linking to `Object` 
```

Comment:
It does not make sense for me that this commet is here. Also We could assign common parts of the graph to value so we do not duplicated it (since we are sorting it at the end anyway)

Reply:
This comment is valid since we have no bigger depth of graph than 1 even though it is not true. Ripping common parts in my opinion it tedious job because there is not so many parts that are common (sometimes it differs on applied type). I will change to `Set` though to eliminate sorting